packages/manager/apps/pci-kubernetes/src/pages/new/steps/NodePoolStep.component.tsx

```diff
@@ -0,0 +1,305 @@
+        flavor: null,
+        name: '',
+        isTouched: false,
+      }));
+    }
+  }, [nodePoolEnabled]);
+
+  return (
+    <>
+      {((!stepper.node.step.isLocked && nodePoolEnabled) ||
+        !nodePoolEnabled) && (
+        <NodePoolToggle
+          nodePoolEnabled={nodePoolEnabled}
+          step={stepper.node.step}
+          onNodePoolEnabledChange={setNodePoolEnabled}
+        />
+      )}
+      <div className="bo border-">
+        {!stepper.node.step.isLocked && nodePoolEnabled && (
+          <>
+            <div className="mb-8">
+              <NodePoolName
+                onTouched={(isTouched: boolean) =>
+                  setNodePoolState((state) => ({ ...state, isTouched }))
+                }
+                hasError={hasError}
+                onNameChange={(name: string) =>
+                  setNodePoolState((state) => ({ ...state, name }))
+                }
+                name={nodePoolState.name}
+              />
+            </div>
+            <div className="mb-8">
+              <NodePoolType
+                projectId={projectId as string}
+                region={stepper.form.region.name}
+                onFlavorChange={setFlavor}
+              />
+            </div>
+            <div className="mb-8">
+              <NodePoolSize
+                isMonthlyBilled={isMonthlyBilled}
+                onScaleChange={(scaling: AutoscalingState) =>
+                  setNodePoolState((state) => ({ ...state, scaling }))
+                }
+                antiAffinity={nodePoolState.antiAffinity}
+              />
+            </div>
+            <div className="mb-8">
+              <NodePoolAntiAffinity
+                isChecked={nodePoolState.antiAffinity}
+                isEnabled={!nodePoolState.scaling?.isAutoscale}
+                onChange={(antiAffinity: boolean) =>
+                  setNodePoolState((state) => ({ ...state, antiAffinity }))
+                }
+              />
+            </div>
+            <div className="mb-8">
+              <BillingStep
+                price={price.hour}
+                monthlyPrice={price.month}
+                monthlyBilling={{
+                  isComingSoon: isPricingComingSoon ?? false,
+                  isChecked: isMonthlyBilled,
+                  check: setIsMonthlyBilled,
+                }}
+                warn={
+                  (nodePoolState.scaling?.isAutoscale && isMonthlyBilled) ??
+                  false
+                }
+              />
+            </div>
+          </>
+        )}
+        {!stepper.node.step.isLocked && nodePoolEnabled && (
+          <OsdsButton
+            variant={ODS_BUTTON_VARIANT.stroked}
+            className="my-6 w-fit"
+            disabled={!nodes || !nodePoolState.scaling || !flavor || undefined}
+            size={ODS_BUTTON_SIZE.sm}
+            color={ODS_TEXT_COLOR_INTENT.primary}
+            {...(isButtonDisabled ? { disabled: true } : {})}
+            onClick={() => {
```

Comment:
This is very hard to read into the JSX part, please move it into a dedicated function